### PR TITLE
Make field optional for the Megastek protocol

### PR DESCRIPTION
**Pull request by Thiritin**
See: https://www.traccar.org/forums/topic/issue-with-megastek-protocol-missing-field/#post-63050

### CODE REVIEW

**Comment by tananaev** [comment]
Looks like CI checks failed. Please fix.

**Comment by Thiritin** [comment]
> Looks like CI checks failed. Please fix.

Yep, thanks, working on it. I'm not really a java dev so I might need some tries.

**Comment by Thiritin** [comment]
> Looks like CI checks failed. Please fix.

I think the test failed because of unreal gps data, I replaced that now please run the tests again.

**Comment by Thiritin** [comment]
Sorry for the failing Test, I will install IntelliJ locally and try to fix it before resubmitting :-) Thank you for your patience!

**Comment by Thiritin** [comment]
@tananaev tests should be green now. Please run again.

File: src/main/java/org/traccar/protocol/MegastekProtocolDecoder.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ private Position decodeOld(Channel channel, SocketAddress remoteAddress, String
             .number("(d+.d+),")                  // hdop
             .number("(d+.d+)?,")                 // speed
             .number("(d+.d+)?,")                 // course
-            .number("(-?d+.d+),")                // altitude
+            .number("(-?d+.d+)?,")                // altitude
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/MegastekProtocolDecoder.java:249]: Please align comment here.

File: src/test/java/org/traccar/protocol/MegastekProtocolDecoderTest.java [hunk 2]
```diff
@@ -39,6 +39,9 @@ public void testDecode() throws Exception {
         verifyPosition(decoder, text(
                 "$MGV002,869152024446923,869152024446923,S,240816,151631,A,5053.83335,N,00424.05702,E,00,10,00,0.88,2.645,76.09,22.7,,206,01,07D1,6600,28,,,,,,,,,01,100,Timer;!"));
 
+        verifyPosition(decoder, text(
+                "0143$MGV002,869152024261564,,R,220621,120804,V,5152.09429,N,01051.32158,E,00,00,00,99.9,,,,,232,10,A5AE,5A4908,22,0000,0000,0,,,,,,00,093,Timer,,;!"));
```
Comment (tananaev) [inline src/test/java/org/traccar/protocol/MegastekProtocolDecoderTest.java:43]: Is this a real sample you got from your device?

Comment (Thiritin) [reply src/test/java/org/traccar/protocol/MegastekProtocolDecoderTest.java:43]: This is a real sample I got from my device yes. I anonymized certain details like coordinates though. I also tested it on my instance and it worked flawlessly.

File: src/main/java/org/traccar/protocol/MegastekProtocolDecoder.java (hunk 1 above)
Comment (Thiritin) [reply src/main/java/org/traccar/protocol/MegastekProtocolDecoder.java]: 309218f2832b6395f9334816e02d755c854f15ad

**Comment by tananaev** [comment]
Merged, thank you.

**Pull request merged**